cmake/kokkos_compiler_id.cmake

```diff
@@ -1,6 +1,5 @@
 kokkos_cfg_depends(COMPILER_ID NONE)
 
-set(KOKKOS_CXX_COMPILER ${CMAKE_CXX_COMPILER})
```

Comment:
Wasn't this guy potentially accessible downstream with the intend to check what Kokkos was build with?